Open Judge System/Web/OJS.Web/Areas/Administration/Views/Contests/ChangeActiveParticipantsEndTime.cshtml

```diff
@@ -109,86 +81,121 @@
+
+                var submitFormButton = $('#submit-form-button');
+                var action = $('#action');
+                var minutes = $('#minutes');
+                var preposition = $('#preposition');
+                var createdBefore = $('#createdBefore');
+                var createdAfter = $('#createdAfter');
+                var usernamePlaceholder = $('#username-placeholder');
+                var headerStrongElements = confirmPopUp.find('h3 strong');
+
+                if (inputValue >= 0) {
+                    action.text('@Resource.Add_are_you_sure');
+                    headerStrongElements.css('color', "#13987E");
+                    preposition.text('@Resource.To');
+                } else {
+                    action.text('@Resource.Subtract_are_you_sure');
+                    headerStrongElements.css('color', "#DF2E1B");
+                    preposition.text('@Resource.From');
+                }
 
-            var addButtonSelector = '#add-time-for-participants-btn';
-            var subtractButtonSelector = '#subtract-time-for-participants-btn';
+                createdBefore.text(createdBeforeValue);
+                createdAfter.text(createdAfterValue);
+                usernamePlaceholder.text(participantUsername);
+                minutes.text(Math.abs(inputValue));
+
+                $('#username-confirmation-input').on('input',
+                    function () {
+                        var value = $(this).val();
+                        if (value === '@this.User.Identity.Name') {
+                            submitFormButton.attr('disabled', false);
+                        } else {
+                            submitFormButton.attr('disabled', true);
+                        }
+                    });
+
+                confirmWindow.show();
+                changeTimeContainer.css('opacity', 0.3);
+
+                submitFormButton.click(
+                    function () {
+                        $('form input[name=TimeInMinutes]').val(timeInput.val());
+  
+                        if ($('#change-time-by-interval').is(':visible')) {
+                            $('#change-time-by-interval-form').submit();
+                        } else if ($('#change-time-by-user').is(':visible')) {
+                            $('#change-time-by-user-form').submit();
+                        }
+                    });
+
+                $('#cancel-btn-confirm-pop-up').click(
+                    function () {
+                        confirmWindow.hide();
+                        changeTimeContainer.css('opacity', 1);
+                    });
+            });
+
+            function setConfirmWindowHeader() {
+                var confirmWindowHeader = $('#confirm-window-header');
+                var headerPrefix = '@Resource.Confirmation_message_header_prefix ';
+                var headerText = '';
+
+                if ($('#change-time-by-interval').is(':visible')) {
```

Comment:
If the above is done with a CSS class, this check can be done less hacky

Reply:
The above CSS class is for another container